roles/container-engine/containerd/tasks/main.yml

```diff
@@ -118,7 +118,7 @@
     mode: 0755
     recurse: true
   with_dict: "{{ containerd_insecure_registries }}"
-  when: containerd_insecure_registries is defined
+  when: use_config_path is defined and use_config_path == true and containerd_insecure_registries is defined
```

Comment:
replace the 
`use_config_path is defined and use_config_path == true`
to
`use_config_path is defined and use_config_path|bool`

may better